### PR TITLE
Hp data

### PR DESCRIPTION
**Pull request by hunterphillips**
# Description
Adds PaymentType Model and Controller, adds functionality to CLI prompt chain

## Related Ticket(s)
#3 

## Problem to Solve
Enables user to add a payment type to an active customer

## Proposed Changes
Changes PaymentType data to include limited selection of generic payment 'types'.
Adds case 3 to switch statement in ui.js

## Expected Behavior
User should be able to add an entry to the payment_types table or is sent back to the main menu if no customer is active

## Steps to Test Solution
```npm run db:reset```
```bangazon```

- successfully choose an active customer
- select Create a payment option
- follow the steps to add a payment method
_invalid input should be denied_
_customer id of 8 is currently out of range, fix coming from Tim_

## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[ ] I added documentation for any new classes/methods

## Deploy Notes

### CODE REVIEW

**Review by iiimosley — APPROVED** [review]
🍔

File: app/controllers/addPaymentTypeCtrl.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        description: `Please enter one of the following payment methods (AmEx, MasterCard, Visa, PayPal)`,
+        type: 'string',
+        pattern: /^AmEx|MasterCard|Visa|PayPal$/,
+        message: 'Please enter one of the available options exactly as it appears',
+        required: true
+      },
+      {
+        name: "accountNumber",
+        description: `Enter account number`,
+        type: 'integer',
```
Comment (Rian501) [inline app/controllers/addPaymentTypeCtrl.js]: When I try to input a long string of numbers (like the length of a credit card number) I am told it must be an integer. When I put in a shorter number ('222') it accepted it. Look into types a bit to see if you can get it to accept bigger integers.

**Comment by Rian501** [comment]
Remember to write tests only for the logic of your functions - they shouldn't be dependent on user's input. If the function needs input to run, supply that input within the test.  🦆

File: app/controllers/addPaymentTypeCtrl.js (hunk 1 above)
Comment (hunterphillips) [reply app/controllers/addPaymentTypeCtrl.js]: Niceeeee, changed 'type' to 'number' in prompt

**Comment by hunterphillips** [comment]
@Rian501 Were you referring to the tests specifically in this branch? In my addPaymentType test, I provide a dummy object for the function to take in. I might be misunderstanding your directions.

**Comment by Rian501** [comment]
@hunterphillips  Not specifically, just saw there was a pending test and made assumptions. 👎  to making assumptions of course.

File: app/models/AddPaymentType.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+const { Database } = require('sqlite3').verbose();
+const path = require('path');
+const db = new Database(path.join(__dirname, '../../', 'bangazon.sqlite'));
+
+module.exports.addCustomerPaymentType = ({ id }, { payment, accountNumber }) => {
+  return new Promise((resolve, reject) => {
+    db.run(`INSERT INTO payment_types
+    VALUES (null, ${id}, "${payment}", ${accountNumber})
+    `,
+    function (err, newPayType) {
```
Comment (TimAConner) [inline app/models/AddPaymentType.js:11]: What is newPayType being used for?  Can't find it being referenced again.

File: app/ui.js [hunk 3]
```diff
@@ -62,6 +64,21 @@ const mainMenuHandler = (err, { choice }) => {
       });
       break;
     }
+
+    // Add Payment Type
+    case 3: {
+      //check if active customer
+      if(getActiveCustomer().id){
```
Comment (TimAConner) [inline app/ui.js:71]: This can later be change to isActiveCustomerSet()

**Review by TimAConner — APPROVED** [review]
Looks good!  All works.

**Pull request merged**